src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -349,6 +352,25 @@ def children_plan_nodes(self) -> List[Union["Selectable", SnowflakePlan]]:
             else []
         )
 
+    def replace_child(self, old_node, new_node) -> None:
+        """This method is called during optimization stage to cut plan tree at a certain node.
+
+        It must only be called on a deep copied plan node, otherwise it will raise an exception.
+        """
+        if not self._is_deep_copied:
```

Comment:
i don't think we need this check, kind of wired. If user want to do an in-place update of child, they can still do it, the job here is to make sure the child update is down correctly.

Reply:
I was talking to Afroz about it. When I was looking at this PR, I was not aware of that this method is only used to update the plan after deepcopy, so I feel in-place update is kinda dangerous, and it shouldn't be used. But if this is only used for compilation stage on plan that gets deep copied, then it's fine. So just to prevent developer from calling this function to update snowflake plan in other places accidentally, I feel a warning or such a check is necessary. If you think it's too much, maybe just a comment (like the one above in docstring) is enough.